Viktoriia_Kovalenko_JavaRush_TODO

cladyland opened the pull request:
just for test..

cladyland committed:
gitignore && hibernate.properties created

cladyland committed:
User entity and classes for working with it created

cladyland committed:
application listener and servlets for login/registration created;
index.jsp, register.jsp and enum for jsp paths created;
pom.xml edited

cladyland committed:
Task entity and classes for working with it created, the logic of the program has been changed according to the new entity

cladyland committed:
The function of editing and deleting tasks and the possibility of logging into the registered account has been added

cladyland committed:
Reduced the number of requests to the database, fixed the task update bug, added a filter for TaskServlet

cladyland committed:
A bug with the display of request details in the address bar has been fixed

cladyland committed:
Tag entity and classes for working with it created, added mappers and made changes to the operation of application services and repositories methods using mappers

cladyland committed:
Added interfaces for services, gitignore edited

cladyland committed:
code connectivity was reduced by changing the signature of services methods, as well as the logic of those methods;
added functionality for working with task tags;
added new methods in the repositories;
creation of a new task is carried out in a separate servlet and jsp file

cladyland committed:
command classes have been added, code work changes have been made in services, mappers and servlets;
password hashing moved in a separate class

cladyland committed:
added the functionality of creating new tags, as well as adding tags to new tasks

cladyland committed:
task statuses and priorities have been added, functionality has been implemented to add these parameters to a new task and display this information to the user

cladyland committed:
Comment entity and classes for working with it created;
added constants for application attributes

cladyland committed:
fixed a bug with the project build

cladyland committed:
added a default implementation for the mapToDTO(Task) method in the TaskMapper interface;
the types of variables in the TaskDTO class have been changed, the redundant variable in the TaskCommand class has been removed

cladyland committed:
the functionality of adding comments has been implemented;
constants for links to servlets have been created

cladyland committed:
separate interfaces have been created for each repository;
redundant classes have been removed

cladyland committed:
SecurityFilter, exception classes, logger configuration added;
fixed a bug with deleting tasks containing comments

cladyland committed:
LoginFilter created;
added validation and handling of invalid data during user authorization

cladyland committed:
RegisterFilter, RegisterException and error.jsp created;
added client-side response status check and message output in case of blank fields

cladyland committed:
added data verification during user registration;
jsp files moved to new packages

cladyland committed:
added additional checks during user registration and authorization, the logs associated with these functions have changed;
new attribute constants have been added

cladyland committed:
additional filters were created, the logic of dependent servlets was partially changed;
 control of access to application resources was strengthened;
 the "not found" page and new constants were added

cladyland committed:
CUTaskFilter created;
task update functionality has been implemented;
additional checks have been created at the request filtering stage;
adding tags to the HTTP session has been moved and distributed between TaskFilter and LoginServlet

cladyland committed:
CommentFilter created;
removed unused methods;
added checks before saving a new comment

cladyland committed:
empty lines at the end of classes have been removed;
a bug with obtaining a username in NewTagServlet has been fixed

cladyland committed:
added new empty lines at the end of java files (so that github does not quarrel)

cladyland committed:
a common jsp file was created for most pages;
added styles, created additional javascript methods

cladyland committed:
the appearance of the pages has been changed;
page new_tag replaced by tag;
added confirmation when deleting a task

cladyland committed:
added password complexity check;
the display of comments on the task_info page has been changed

cladyland committed:
a file with an sql script and a configuration file for launching the application have been created

cladyland committed:
war builded

cladyland committed:
web.xml edited

cladyland committed:
Create README.md

antonkupreychik commented on pom.xml:
```diff
@@ -7,29 +7,27 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <org.mapstruct.version>1.5.3.Final</org.mapstruct.version>
+        <org.projectlombok.version>1.18.24</org.projectlombok.version>
     </properties>
```
опять же, надо решить, где ты пишешь версии своих зависимостей, или все там или там)

antonkupreychik commented on src/main/java/kovalenko/vika/filter/CommentFilter.java:
```diff
@@ -0,0 +1,74 @@
+        LOG.debug("'CommentFilter' is destroyed");
+    }
+
+    private void notfoundRedirect(HttpServletResponse response, String parameterName) throws IOException {
```
notFound

antonkupreychik commented on src/main/java/kovalenko/vika/filter/RegisterFilter.java:
```diff
@@ -0,0 +1,118 @@
+        request.setAttribute(USERNAME, parameters.get(USERNAME));
+    }
+
+    private void nullChecking(HashMap<String, String> parameters) {
```
вот этот метод бросит NullPointer. В логах ты это увидишь, а вот пользователь никак не поймет, что он что-то не ввел. Как правило, обычно это делается таким образом: если валится валидация, то ее нужно выводить пользователю на экран, сделать это можно примерно так - https://www.w3schools.com/howto/howto_js_alert.asp

antonkupreychik commented on src/main/java/kovalenko/vika/filter/RegisterFilter.java:
```diff
@@ -0,0 +1,118 @@
+        }
+        return countOfBlankParams == 0;
+    }
+
```
такие проверки явно встречаются много где у тебя в проекте, вынеси их в middleware

AndryLat commented on src/main/java/kovalenko/vika/dao/impl/UserDAOImp.java:
```diff
@@ -0,0 +1,54 @@
+import org.slf4j.LoggerFactory;
+
+public class UserDAOImp implements UserDAO {
+    private static final Logger LOG = LoggerFactory.getLogger(UserDAOImp.class);
```
если используем ломбок, то логгер хорошо бы тоже в ломбок.

AndryLat commented on src/main/java/kovalenko/vika/filter/RegisterFilter.java:
```diff
@@ -0,0 +1,118 @@
+
+        if (paramsAreNotBlank(parameters)) {
+            httpRequest.setAttribute(PARAMETERS, parameters);
+            LOG.info(PRE_CHECK, "OK");
```
С такого лога вообще непонятно что тут происходит. немного бы конкретики что именно это за запрос было бы очень круто.

AndryLat commented on src/main/java/kovalenko/vika/mapper/BasicMapper.java:
```diff
@@ -0,0 +1,7 @@
+package kovalenko.vika.mapper;
+
+public interface BasicMapper<T, E> {
```
класс!

AndryLat reviewed (approved):
Крутейший проект! Были бы тесты - это был бы бриллиант!

cladyland committed:
added @Slf4j from lombok instead of direct use Logger and LoggerFactory;
fixed method name in CommentFilter;
added ordering for filters with the same paths

cladyland committed:
constants classes moved to new package;
auxiliary utility classes have been created to reduce code duplication;
added null and blank checking for tasks titles and tags

cladyland committed:
hibernate settings and DB initialization syntax changed to use PostgreSQL

cladyland committed:
Session factory settings are placed in a separate class;
Dockerfile and docker-compose created

cladyland committed:
removed unused setters;
added new logs to repositories and fixed queries

cladyland committed:
UserService logic is divided into two interfaces;
return type of methods which values are not used has been changed to void;
an additional methods was created in TaskMapper and ServletUtil;
created another constructor in RegisterException to modernize the output of error information to the user

cladyland committed:
refactored code for servlets and filters related to tasks, tags and comments

cladyland committed:
refactored code for servlets and filters

cladyland committed:
refactored code for util classes;
log4j2.xml updated

cladyland committed:
updated maximum length for entities elements with varchar type;
scripts and css-styles are moved to separate files;
web.xml updated

cladyland committed:
added new styles;
added links to files with scripts and styles to head.jsp;
index.jsp and register.jsp updated

cladyland committed:
added new styles and logout functionality (frontend and backend);
todo.jsp updated

cladyland committed:
TagFilter created;
added new js functions;
the amount of saving data for a new task has been expanded;
new_task.jsp updated

cladyland committed:
fixed the bug with the order of tasks in the task table after updating a task;
fixed attributes setting bug for the "back" button which is on tag.jsp page;
new styles added;
web pages updated

cladyland committed:
added style for comments;
index.jsp and task_info.jsp updated

cladyland committed:
added styles for error pages;
not_found.jsp and error.jsp updated

cladyland committed:
added functionality to check the content of post-requests for the presence of html and partial replacement of requests parameters

cladyland committed:
tests for classes of package "utils" created;
pom.xml updated;
added one more character to define password complexity

cladyland committed:
tests for CommentServiceImp and RegisterServiceImp created;
auxiliary class Verifier created

cladyland committed:
tests for TagServiceImp, TaskServiceImp and UserServiceImp created

cladyland committed:
abstract class for *ServletTest.java created;
tests for all classes of package "servlet" created

cladyland committed:
tests for classes of package "filter.security" created;
abstract class for *FilterTest.java and abstract common class for filters and servlets created;
test classes of "servlet" package updated

cladyland committed:
tests for classes of package "filter.task" created;
new attribute constants added;
CUTaskFilter, TagFilter, NewTaskServletTest and TagServletTest updated

cladyland committed:
tests for LoginFilter and RegisterFilter created

cladyland committed:
configured H2 in-memory DB for tests and created h2_init.sql;
tests for classes of "dao.impl" package and AppContextListener created;
new constant-classes created;
pom.xml updated

cladyland committed:
the presence of setters in classes fields of package "model" has been changed;
DBConfig.java updated

cladyland committed:
log4j config and Dockerfile updated;
zip archive with files for launch created

cladyland committed:
README.md updated
